app/src/main/java/org/mozilla/rocket/download/DownloadInfoRepository.kt

```diff
@@ -41,21 +39,18 @@ class DownloadInfoRepository {
         }
     }
 
-    fun queryDownloadItems(runningIds: LongArray, listenerList: OnQueryListCompleteListener) {
+    fun queryDownloadingItems(runningIds: LongArray, listenerList: OnQueryListCompleteListener) {
         ThreadUtils.postToBackgroundThread {
-            var cursor: Cursor? = null
-            try {
-                val manager = DownloadInfoManager.getInstance().downloadManager//context.getSystemService(Context.DOWNLOAD_SERVICE) as DownloadManager
-                val query = DownloadManager.Query()
-                query.setFilterById(*runningIds)
-                query.setFilterByStatus(DownloadManager.STATUS_RUNNING)
-                cursor = manager.query(query)
-                if (cursor != null) {
+            val query = DownloadManager.Query()
+            query.setFilterById(*runningIds)
+            query.setFilterByStatus(DownloadManager.STATUS_RUNNING)
+            DownloadInfoManager.getInstance().downloadManager.query(query).use {
```

Comment:
replace "use" with "apply" or "run"

Reply:
Will keep "use" to leverage the auto closable implementation